internal/rootcoord/dml_channels.go

```diff
@@ -17,42 +17,137 @@
 package rootcoord
 
 import (
+	"container/heap"
 	"context"
 	"fmt"
 	"sync"
 
 	"github.com/milvus-io/milvus/internal/metrics"
 
-	"go.uber.org/atomic"
 	"go.uber.org/zap"
 
 	"github.com/milvus-io/milvus/internal/log"
 	"github.com/milvus-io/milvus/internal/mq/msgstream"
 )
 
 type dmlMsgStream struct {
-	ms     msgstream.MsgStream
-	mutex  sync.RWMutex
-	refcnt int64
+	ms       msgstream.MsgStream
+	mutex    sync.RWMutex
+	chanName string
+	refcnt   int64 // current in use count
+	used     int64 // total used counter in current run, not stored in meta so meant to be inaccurate
+	idx      int64 // idx for name
+	pos      int   // position in the heap slice
+}
+
+// RefCnt returns refcnt with mutex protection.
+func (dms *dmlMsgStream) RefCnt() int64 {
+	dms.mutex.RLock()
+	defer dms.mutex.RUnlock()
+	return dms.refcnt
+}
+
+// RefCnt returns refcnt with mutex protection.
+func (dms *dmlMsgStream) Used() int64 {
+	dms.mutex.RLock()
+	defer dms.mutex.RUnlock()
+	return dms.used
+}
+
+// AddUsage increases refcnt and used.
+func (dms *dmlMsgStream) AddUsage() {
+	dms.mutex.Lock()
+	defer dms.mutex.Unlock()
+	dms.refcnt++
+}
+
+// AddUsage increases used.
+func (dms *dmlMsgStream) BookUsage() {
+	dms.mutex.Lock()
+	defer dms.mutex.Unlock()
+	dms.used++
+}
+
+// RemoveUsage decreases refcnt only.
+func (dms *dmlMsgStream) RemoveUsage() {
```

Comment:
function name is confusing, because the API name is "RemoveUsage", but in the code "refcnt" is changed

Reply:
Fixed.